cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java

```diff
@@ -291,6 +293,32 @@ private void scheduleJob(TriggerKey triggerKey, String scheduleName, String cron
     return getScheduledRuntime(program, programType, false);
   }
 
+  public List<ScheduledRuntime> nextScheduledRuntime(ProgramId program, SchedulableProgramType programType,
+                                                     long startTimeSecs, long endTimeSecs)
+    throws SchedulerException {
+    // decrease the start time by one second to include the next fire time that is equal to the start time
+    Date startTime = new Date(TimeUnit.SECONDS.toMillis(startTimeSecs - 1));
```

Comment:
you just have to reduce by 1 millisecond

Reply:
the smallest resolution in Quartz is second and inside Quartz I saw code like this
```
@Override
    public Date getFireTimeAfter(Date afterTime) {
...
        if (getStartTime().after(afterTime)) {
            afterTime = new Date(getStartTime().getTime() - 1000l);
        }
...```